### PR TITLE
auth: depend on nodeIDHandler directly 

### PR DESCRIPTION
**Pull request by mhofstetter**
After removing the node IDs from the IPCache with #27029, the auth module should directly depend on the NodeIDHandler instead of the IPCache.

This PR refactors the dependency from the IPCache to the NodeIDHandler.

In addition, the unused NodeIDHandler gets deleted from the IPCache Configuration.

### CODE REVIEW

File: pkg/auth/manager_test.go [hunk 1]
```diff
@@ -175,6 +176,13 @@ type fakeIPCache struct {
 	nodeIdMappings map[uint16]string
 }
 
+func (r *fakeIPCache) DumpNodeIDs() []*models.NodeID {
```
Comment (gandro) [inline pkg/auth/manager_test.go]: nit: Consider renaming this type to `fakeNodeIDHandler`

**Review by gandro — APPROVED** [review]
Excellent! Thanks a lot for this!

**Review by meyskens — APPROVED** [review]
Apart from the nit already noted this is an LGTM for me!

**Review by christarazi — APPROVED** [review]

File: pkg/auth/manager_test.go (hunk 1 above)
Comment (mhofstetter) [reply pkg/auth/manager_test.go]: @meyskens thanks a lot for dealing with that during my PTO!

**Pull request merged**